src/module.cpp

```diff
@@ -536,8 +536,43 @@ void Module::AddGlobalVariable(const std::string &name, const Type *type, Expr *
                 // the same type as the global.  (But not if it's an
                 // ExprList; they don't have types per se / can't type
                 // convert themselves anyway.)
-                if (llvm::dyn_cast<ExprList>(initExpr) == nullptr)
+                if (llvm::dyn_cast<ExprList>(initExpr) == nullptr) {
                     initExpr = TypeConvertExpr(initExpr, type, "initializer");
+                } else {
+                    ExprList *exprList = llvm::dyn_cast<ExprList>(initExpr);
+
+                    // If we have exprList that initalizes const varying int, e.g.:
+                    // static const int x = { 0, 1, 2, 3 };
+                    // then we need to convert initExpr to constExpr that can
+                    // be utilized later in arithmetic expressions that can be
+                    // calculated in compile time (Optimize function call below),
+                    if (type->IsConstType() && type->IsVaryingType()) {
+                        const int N = g->target->getVectorWidth();
+                        int32_t *vals = (int32_t *)alloca(N * sizeof(int32_t));
```

Comment:
Why not to use `std::vector` instead of manually managing memory?
`std::vector<int32_t> vals(N, 0);`

Reply:
Done